### DISCUSSION TITLE
Octicons replacement

### DISCUSSION THREAD

**Discussion by Daeraxa** [pulsar-edit/.github#86] [Pulsar Features]
Currently Pulsar is still using a really old version (4.4.0) of Octicons as its built in icon font which is a bit limited compared to modern and current options (including Octicons which is now on v17.9.0).

A couple of options:
- Upgrade to a newer Octicons version (lots of breaking changes so possibly not just a simple upgrade).
- Use a different font family (Font Awesome, Material design, Feather etc.)
- Use an aggregated set like the Nerd Font symbols (https://www.nerdfonts.com/) which has a whole bunch of them included.

We would obviously need to keep the original Octicon compatibility but I think additional flexibility and choice of icons for community packages would be quite useful.

**Comment by mauricioszabo** [comment]
I agree with the idea, and I actually was thinking on doing this soon-ish. There's a package called "StyleGuide" that basically have the icons and their respective classes expected on the editor, could be a good start :)

**Comment by kaosine** [comment]
I'd definitely go with FA or something instead imo. Esp if we can get system icons and such and just use it across all repos and not just the main one.

**Comment by confused-Techie** [comment]
So older topic I know.

But was just looking around in the code base, and I honestly thinking doing an upgrade to the octicons would be simpler than we original thought. Pulsar already bundles some icons from Octicons 2.1.2 and includes the newer ones.

So we could upgrade, and for any of the icons that no longer exist under the same name keep these, and replace the others. Sure some icons would change slightly, but wouldn't imagine they would break, since it'd just be an updated icon.

Even further Pulsar has Octicons 2.1.2 and Octicons 4.4.0, and it's own AtomIcons installed.

I think there's real potential for us just going as overboard as we'd like, updating to newer Octicons and if we wanted implementing something like Font Awesome, Material Design, Feather. There's no reason we can't just go crazy. Other than space and complexity for no good reason.

But I'd be happy to try and take a stab at updating our implementation of Octicons first, and see how it goes if we'd like

**Reply by mauricioszabo** [reply]
You have our blessing :D

**Comment by confused-Techie** [comment]
My [PR](https://github.com/pulsar-edit/pulsar/pull/509) is up to upgrade octicons!
